src/Form/EventsAddEditCallbacks.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\localgov_events\Form;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\Entity\Node;
+
+/**
+ * Callbacks and handlers for Event node add and edit forms.
+ */
+class EventsAddEditCallbacks {
+
+  /**
+   * Text displayed when using the location from the venue.
+   *
+   * @var string
+   */
+  const LOCATION_FROM_VENUE = '<p>Using location from venue. To use a different location remove any existing locations and add a new one below.</p>';
+
+  /**
+   * Text displayed when not using the location from the venue.
+   *
+   * @var string
+   */
+  const NOT_LOCATION_FROM_VENUE = '<p>Using a different location from venue. To use the same location remove the location below and save the page.</p>';
+
+  /**
+   * Return location field widget.
+   *
+   * @param array $form
+   *   Form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form data.
+   *
+   * @return array
+   *   Location render array to display in form.
+   */
+  public static function venueChangeCallback(array &$form, FormStateInterface $form_state) {
+    $location_element = $form['localgov_event_location'];
+    $venue_value = $form_state->getValue('localgov_event_venue');
+    if (isset($venue_value[0]['target_id']) && $venue_value[0]['target_id']) {
+
+      // This could be used for automatically loading the location Geo object
+      // into the node add/edit form when the venue is changed.
+      // For now we just set a message as this is handled on form submit.
+      $location_element['#prefix'] = self::LOCATION_FROM_VENUE;
+    }
+
+    return $location_element;
+  }
+
+  /**
+   * Submit handler for the add Event node form.
+   *
+   * @param array $form
+   *   Form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form data.
+   */
+  public static function submitHandler(array &$form, FormStateInterface $form_state) {
+    // Set location if there's a venue but no location.
+    $venue_value = $form_state->getValue('localgov_event_venue');
+    $location_value = $form_state->getValue('localgov_event_location');
+    if (
+      isset($venue_value[0]['target_id']) &&
+      $venue_value[0]['target_id'] &&
+      !isset($location_value[0]['target_id'])
+    ) {
+      $event = $form_state->getFormObject()->getEntity();
+      $venue = Node::load($venue_value[0]['target_id']);
+      $location_value = $venue->get('localgov_location')->getValue();
+      $event->set('localgov_event_location', $location_value);
+      $event->save();
+    }
+  }
+
+  /**
+   * Configure node add / edit form.
+   *
+   * @param array $form
+   *   Form array.
+   */
+  public static function configureNodeForm(array &$form) {
+    if (isset($form['localgov_event_venue'])) {
+
+      // Add submit and Ajax callbacks.
+      $form['actions']['submit']['#submit'][] = 'Drupal\localgov_events\Form\EventsAddEditCallbacks::submitHandler';
```

Comment:
You can also consider ```'::submitHandler'```

@see https://www.drupal.org/docs/drupal-apis/javascript-api/ajax-forms#s-adding-the-ajax-event

Reply:
So this is not actually the forms parent class. It's more of a namespace for functions related to the events add and edit forms functionality; it's prevents cluttering the .module file. So I don't think leaving out the fully namespaced class is going to work ehere.